File: Incubation/trunk/psim/src/org/space/oss/psim/PSim.java
package org.space.oss.psim;

import java.util.Properties;

public interface PSim 
{
	public void init(Config cfg);
    public void shutdown();
    public Properties getVersionInfo();
	public void save(String dir);
	public void load(String dir);
    
    public CommandService getCommandService();
    public TelemetryService getTelemetryService();
    public SpacecraftService getSpacecraftService();
    public TimeService getTimeService();
}

File: Incubation/trunk/psim-scala/src/main/scala/org/space/oss/psim/PSim.scala
package org.space.oss.psim

import java.text.SimpleDateFormat
import java.util.TimeZone

trait PSim 
{
	def init(cfg: Config)
	def getSpacecraftService: SpacecraftService
	def getTimeService: TimeService
}

object PSimUtil {
 	 val formatter:SimpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z")
 	 formatter.setTimeZone(TimeZone.getTimeZone("GMT"))
 	 
 	 def setTimeZone(tz:TimeZone) = formatter.setTimeZone(tz)
	 def formatTime(t:Long) = formatter.format(t) 
	 def parseTime(timeStr: String) = formatter.parse(timeStr)
}